llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -8074,14 +8105,20 @@ AArch64TargetLowering::LowerCall(CallLoweringInfo &CLI,
                                  PStateSM, false);
   }
 
+  if ((RequiresLazySave && !CalleeAttrs.preservesZA()) || ToggleZA)
```

Comment:
Sorry for not thinking ahead on this one, but seeing this I just wondered if it would be easier to have two interfaces:
* `CallerAttrs.requiresDisablingZABeforeCall(CalleeAttrs)`
* `CallerAttrs.requiresEnablingZAAfterCall(CalleeAttrs)`

instead of `requiresZAToggle`?